vault/resource_terraform_cloud_secret_backend.go

```diff
@@ -0,0 +1,246 @@
+		Create: terraformCloudSecretBackendCreate,
+		Read:   terraformCloudSecretBackendRead,
+		Update: terraformCloudSecretBackendUpdate,
+		Delete: terraformCloudSecretBackendDelete,
+		Exists: terraformCloudSecretBackendExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Default:     "terraform",
+				Description: "Unique name of the Vault Terraform Cloud mount to configure",
+				StateFunc: func(s interface{}) string {
+					return strings.Trim(s.(string), "/")
+				},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old+"/" == new || new+"/" == old
+				},
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Specifies the Terraform Cloud access token to use.",
+				Sensitive:   true,
+			},
+			"address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "https://app.terraform.io",
+				Description: "Specifies the address of the Terraform Cloud instance, provided as \"host:port\" like \"127.0.0.1:8500\".",
+			},
+			"base_path": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "/api/v2/",
+				Description: "Specifies the base path for the Terraform Cloud or Enterprise API.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Human-friendly description of the mount for the backend.",
+			},
+			"default_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     "0",
+				Description: "Default lease duration for secrets in seconds",
+			},
+			"max_lease_ttl_seconds": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Default:     "0",
+				Description: "Maximum possible lease duration for secrets in seconds",
+			},
+		},
+	}
+}
+
+func terraformCloudSecretBackendCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	path := d.Get("path").(string)
+	address := d.Get("address").(string)
+	token := d.Get("token").(string)
+	basePath := d.Get("base_path").(string)
+
+	configPath := terraformCloudSecretBackendConfigPath(path)
+
+	info := &api.MountInput{
+		Type:        "terraform",
+		Description: d.Get("description").(string),
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%ds", d.Get("default_lease_ttl_seconds")),
+			MaxLeaseTTL:     fmt.Sprintf("%ds", d.Get("max_lease_ttl_seconds")),
+		},
+	}
+
+	d.Partial(true)
+	log.Printf("[DEBUG] Mounting Terraform Cloud backend at %q", path)
+
+	if err := client.Sys().Mount(path, info); err != nil {
+		return fmt.Errorf("Error mounting to %q: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Mounted Terraform Cloud backend at %q", path)
+	d.SetId(path)
+
+	d.SetPartial("path")
```

Comment:
`SetPartial` is deprecated, I believe we should just use `d.Set` here and set the values

- https://github.com/hashicorp/terraform-plugin-sdk/blob/9d2850173e33bf6e4b4fa0fd0819f602b45e23c8/helper/schema/resource_data.go#L206-L215

Reply:
SetPartial removed